TS_aqa_course_homework/ts_hw15/task2.ts

```diff
@@ -20,4 +20,62 @@
       Реализовывать метод getDetails(), который должен выводить полную информацию об объекте вида:
         'My name is Elena TSovna, I am software developer with 6 years of experience in TypeScript and 6000$ salary' (пример для девелопера) */
 
-        
\ No newline at end of file
+interface IPerson {
+  name: string;
+  surname: string;
+  experienceYears: number;
+  getDetails(): string;
+}
+
+abstract class Employee implements IPerson {
+  protected salary: number = 0;
+  constructor(
+    public name: string,
+    public surname: string,
+    public experienceYears: number,
+  ) {
+    this.calculateSalary();
+  }
+  protected abstract calculateSalary(): void;
+  abstract getDetails(): string;
+}
+
+class Manager extends Employee {
+  constructor(
+    public name: string,
+    public surname: string,
+    public experienceYears: number,
```

Comment:
Свойства, что передаются в `super` пишем без модификаторов доступа, иначе мы создаём текущему классу собственное свойство

```suggestion
     name: string,
     surname: string,
     experienceYears: number,
```